staging/src/k8s.io/kubectl/pkg/cmd/debug/profiles.go

```diff
@@ -49,11 +62,195 @@ func profileLegacy(pod *corev1.Pod, containerName string, target runtime.Object)
 			})
 		}
 
-		pod.Spec.HostIPC = true
-		pod.Spec.HostNetwork = true
-		pod.Spec.HostPID = true
+		setHostNamespace(pod, true)
 		return nil
 	default:
 		return fmt.Errorf("the %s profile doesn't support objects of type %T", ProfileLegacy, target)
 	}
 }
+
+func isPodCopy(pod *corev1.Pod, target runtime.Object) bool {
+	if asserted, ok := target.(*corev1.Pod); !ok {
+		return false
+	} else {
+		return pod != asserted
+	}
+}
+
+type debugStyle int
+
+const (
+	styleEphemeral debugStyle = iota
+	stylePodCopy
+	styleNode
+)
+
+func getDebugStyle(pod *corev1.Pod, target runtime.Object) debugStyle {
+	switch target.(type) {
+	case *corev1.Pod:
+		if isPodCopy(pod, target) {
+			return stylePodCopy
+		}
+		return styleEphemeral
+	case *corev1.Node:
+		return styleNode
+	}
+	return debugStyle(-1) // unknown
```

Comment:
Given that the result is consumed in a `switch` statement, returning multiple values including an `error` values makes the downstream consuming semantics a little less ergonomic. It would also lead to a `default` case that would not be reachable or a trailing return statement after the switch which is not reachable.

Reply:
This style looks not Go style. It should return error, if case is not found.